test/runtests.jl

```diff
@@ -63,3 +63,28 @@ end
     @test_throws ArgumentError("unknown algorithm BILL") Algorithm(:BILL)
     @test_throws ArgumentError("unknown algorithm BILL") Opt(:BILL, 420)
 end
+
+@testset "Failure to converge shouldn't error" begin
+    function rosenbrock(x, G)
+        if length(G) > 0
+            G[1] = -2 * (1 - x[1]) - 400 * x[1] * (x[2] - x[1]^2)
+            G[2] = 200 * (x[2] - x[1]^2)
+        end
+        (1 - x[1])^2 + 100 * (x[2] - x[1]^2)^2
+    end
+    function circ_cons(res, x, J)
+        res[1] = x[1]^2 + x[2]^2 - 1.0
+        if length(J) > 0
+            J[1, 1] = 2x[1]
+            J[2, 1] = 2x[2]
+        end
+    end
+
+    opt = Opt(:AUGLAG, 2)
+    opt.local_optimizer = Opt(:LD_LBFGS, 2)
+    opt.min_objective = rosenbrock
+    NLopt.equality_constraint!(opt, circ_cons, [1e-8])
+    (minf, minx, ret) = optimize(opt, [0.5, 0.5])
```

Comment:
Should we add a test for the `ret`?

Reply:
Added it, but I am not sure why it gives failure

Reply:
What do you mean? Isn't the point of the test that it's a `FAILURE`?

Reply:
I meant that given the results the `FAILURE` is suspicious. You are correct that the point of this test is that in such cases we shouldn't error but return the result.

Reply:
The cause is probably the nonlinear equality constraint (which is non convex)?

Reply:
Yeah but the constraint residual is below the set tolerance

```
julia> 1.0 -sum(abs2, minx)
6.695932697198259e-11
```

Reply:
Not sure then

Reply:
yeah, but that's just why this makes sense